migrations/versions/660_move_framework_framework_agreement_version.py

```diff
@@ -0,0 +1,37 @@
+"""Move Framework.framework_agreement_version into new framework_agreement_details json field
+
+Revision ID: 660
+Revises: 650
+Create Date: 2016-06-29 11:54:27.719038
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '660'
+down_revision = '650'
+
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+def upgrade():
+    op.add_column('frameworks', sa.Column('framework_agreement_details', postgresql.JSON(), nullable=True))
+    op.execute("""
+        UPDATE frameworks SET
+            -- using to_json to ensure framework_agreement_version gets properly escaped for json
+            framework_agreement_details = (
+                '{"frameworkAgreementVersion":' || to_json(framework_agreement_version) || '}'
+            )::json
+        WHERE
+            framework_agreement_version IS NOT NULL and framework_agreement_version != ''
+    """)
+    op.drop_column('frameworks', 'framework_agreement_version')
+
+
+def downgrade():
+    op.add_column('frameworks', sa.Column('framework_agreement_version', sa.VARCHAR(), autoincrement=False, nullable=True))
+    op.execute("""
+        UPDATE frameworks SET
+            framework_agreement_version = framework_agreement_details->'frameworkAgreementVersion'
```

Comment:
This downgrade doesn't quite work... for some reason the string that gets set after dereferencing `frameworkAgreementVersion` includes double quotes around the string.  I added a version for DOS to my local DB to check I wasn't going mad:
![screen shot 2016-07-06 at 09 44 33](https://cloud.githubusercontent.com/assets/6525554/16612333/58106062-435e-11e6-8eed-8bb6c93ff3c9.png)

I think it needs the `->>` operator to work properly.

Other than that looking good, sorry it's taken so long to look at this.